ansible/modules/hashivault/hashivault_secret_engine.py

```diff
@@ -0,0 +1,175 @@
+             is specified, its value will take precedence"
+        default: to environment variable VAULT_CAPATH
+    client_cert:
+        description:
+            - "path to a PEM-encoded client certificate for TLS authentication to the Vault server"
+        default: to environment variable VAULT_CLIENT_CERT
+    client_key:
+        description:
+            - "path to an unencrypted PEM-encoded private key matching the client certificate"
+        default: to environment variable VAULT_CLIENT_KEY
+    verify:
+        description:
+            - "if set, do not verify presented TLS certificate before communicating with Vault server : setting this
+             variable is not recommended except during testing"
+        default: to environment variable VAULT_SKIP_VERIFY
+    authtype:
+        description:
+            - "authentication type to use: token, userpass, github, ldap, approle"
+        default: token
+    token:
+        description:
+            - token for vault
+        default: to environment variable VAULT_TOKEN
+    username:
+        description:
+            - username to login to vault.
+        default: to environment variable VAULT_USER
+    password:
+        description:
+            - password to login to vault.
+        default: to environment variable VAULT_PASSWORD
+    name:
+        description:
+            - name of secret backend
+    backend:
+        description:
+            - type of secret backend
+    description:
+        description:
+            - description of secret backend
+    config:
+        description:
+            - config of secret backend
+        default: "{'default_lease_ttl': 2764800, 'max_lease_ttl': 2764800, 'force_no_cache': False}"
+    state:
+        description:
+            - state of secret backend. choices: enabled, present, disabled, absent
+    options:
+        description:
+            - Specifies mount type specific options that are passed to the backend. NOT included unless backend == kv
+'''
+EXAMPLES = '''
+---
+- hosts: localhost
+  tasks:
+    - hashivault_secret_engine:
+        name: "ephemeral"
+        backend: "generic"
+'''
+
+def main():
+    argspec = hashivault_argspec()
+    argspec['name'] = dict(required=True, type='str')
+    argspec['backend'] = dict(required=False, type='str', default='')
+    argspec['description'] = dict(required=False, type='str')
+    argspec['config'] = dict(required=False, type='dict', default={'default_lease_ttl': DEFAULT_TTL, 'max_lease_ttl': DEFAULT_TTL, 'force_no_cache': False})
+    argspec['state'] = dict(required=False, type='str', choices=['present', 'enabled', 'absent', 'disabled'], default='present')
+    argspec['options'] = dict(required=False, type='dict', default={'version': '1'})
+    module = hashivault_init(argspec)
+    result = hashivault_secret_enable(module)
+    if result.get('failed'):
+        module.fail_json(**result)
+    else:
+        module.exit_json(**result)
+
+
+@hashiwrapper
```

Comment:
I'm wondering if the document problem is this should be named hashivault_secret_engine

Reply:
this has got to be it. now that 9.2 is released i can enable to test too